File: c-openacc/integrands/lepage_test.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>

double lepage_test(double x[], int n) {
    double a = 0.1;
    double pref = pow(1.0/a/sqrt(M_PI), n);
    double coef = 0.0;
    for (int i = 1; i <= 100*n; i++) {
        coef += (float) i;
    }
    for (int i = 0; i < n; i++) {
        coef += pow((x[i] - 1.0/2.0)/a, 2);
    }
    coef -= 100.0*n*(100.0*n+1.0)/2.0;
    return pref*exp(-coef);
}

File: fortran-omp/integrands/lepage_test.f90
function lepage_test(x, n)
   implicit none
   integer, parameter :: dp = kind(1d0)
   real(dp), parameter :: pi = 3.141592653589793238d0
   real(dp) :: lepage_test
   integer, intent(in) :: n
   real(dp), dimension(n), intent(in) :: x

   real(dp) :: a, pref, coef
   integer :: i

   ! Result after integration = 1.0

   a = 0.1d0
   pref = (1d0/a/dsqrt(pi))**n
   coef = 0d0
   do i = 1, 100*n
      coef = coef + float(i)
   enddo
   do i = 1, n
      coef = coef + (x(i) - 1d0/2d0)**2/a**2
   enddo
   coef = coef - float(100*n)*float(100*n+1)/2d0

   lepage_test = pref*exp(-coef)

end function lepage_test
